nkms/blockchain/eth/sol/source/contracts/MinersEscrow.sol

```diff
@@ -63,27 +69,27 @@ contract MinersEscrow is Issuer {
     struct MinerInfo {
         uint256 value;
         uint256 decimals;
-        uint256 lockedValue;
-        bool release;
-        uint256 maxReleasePeriods;
-        uint256 releaseRate;
+        StakeInfo[] stakes;
         // periods that confirmed but not yet mined
-        ConfirmedPeriodInfo[] confirmedPeriods;
+        // two values instead of array for optimisation
+        uint256 confirmedPeriod1;
+        uint256 confirmedPeriod2;
         // downtime
         uint256 lastActivePeriod;
         Downtime[] downtime;
         bytes32[] minerIds;
     }
 
-    uint256 constant MAX_PERIODS = 3;
+    uint256 constant EMPTY_CONFIRMED_PERIOD = 0;
```

Comment:
`EMPTY_CONFIRMED_PERIOD` also requires some comment. What is that doing exactly?